mkosi/config.py

```diff
@@ -248,6 +249,18 @@ def config_parse_compression(value: Optional[str], old: Optional[Compression]) -
     except KeyError:
         return Compression.zst if parse_boolean(value) else Compression.none
 
+def config_parse_seed(value: Optional[str], old: Optional[str]) -> Optional[str]:
+    if not value:
+        return None
+
+    if value == "random":
+        return value
```

Comment:
If we calculate a random UUID ourselves, we can make the type of `seed` in `MkosiConfig of `UUID` instead of having to store a string.

Reply:
I think the  meaning of no `--seed` flag is different from `random` and different from a specific uuid:

> If not specified it is attempted to read the machine ID from the host (or more precisely, the root directory configured via --root=) and use it as seed instead, falling back to a randomized seed otherwise. Use --seed=random to force a randomized seed. Explicitly specifying the seed may be used to generated strictly reproducible partition tables.

 So i think we either need a type that can hold any valid UUID and `random` or should just use `Optional[str]`.

Reply:
Right but we never want /etc/machine-id to be used in our case since there is none anyway. So always specifiying `--seed` and just making it random if not specified seems like the way to go.

Reply:
Should we even allow `random` as a user chosen value in the mkosi config?
I feel like users of repart could be tempted to call mkosi with `--seed random` so this should still be allowed and could be internally tracked as `None`.

Reply:
That sounds good to me, but make sure to translate `None` back to "random" in the summary then

Reply:
And also pass `--randomize` to systemd-repart if `seed` is `None`